server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -1063,11 +1063,11 @@ public boolean isSegRepEnabled() {
     }
 
     public boolean isSegRepLocalEnabled() {
-        return isSegRepEnabled() && !isSegRepWithRemoteEnabled();
+        return isSegRepEnabled() && !isRemoteStoreEnabled();
     }
 
     public boolean isSegRepWithRemoteEnabled() {
```

Comment:
lets remove this method altogether and also `isSegRepLocalEnabled` as well .

Reply:
then remove them as well .

Reply:
@gbbafna this is being used when the index is only segrep enabled without remote store. Basis this setting, the checkpoint publisher is added in the internal refresh listeners.